src/engine/Events.ts

```diff
@@ -93,8 +93,29 @@ export class GameEvent<T> {
    public target: T;
 }
 
+export abstract class BubblingEvent extends GameEvent<Actor> {
```

Comment:
I think it would be okay, to move this implementation to GameEvent... that way all events can bubble without extra effort.

Reply:
Now i am preparing updated PR, and i have extended functionality of that class. So, i am not moving Merging BubblingEvent to GameEvent to let you check all changes i made. And after revision, if you'll feel like merging this two guys together, just comment and i'll do it. :3

Reply:
I think this looks good, I like that the propagation logic is stored with the event. I'd still like to have this as a feature of all GameEvents, let's move this implementation merge to GameEvent.